test/extended/util/framework.go

```diff
@@ -235,7 +236,7 @@ func WaitForADeployment(client kclient.ReplicationControllerInterface,
 			}
 		}
 	}
-	return fmt.Errorf("the deploy did not finish within 3 minutes")
+	return fmt.Errorf("the deploy did not finish within %v minutes", maxWaitingTime)
```

Comment:
For me this change could go in a PR by itself, it has nothing to do with the other changes.

And look at the pattern we're fixing here: the error message had "3 minutes" hard coded. We're fixing the "3" part, but ignoring the "minutes" part.

IMO the right fix is to make `maxWaitingTime` be a `time.Duration`, and not an `int`:

```go
maxWaitingTime := 15 * time.Minute
// ...
endTime := startTime.Add(maxWaitingTime)
// ...
return fmt.Errorf("the deploy did not finish within %v", maxWaitingTime)
```

Go knows how to print `time.Duration` nicely (it implements `Stringer`).